### PR TITLE
Allow webapp to request legacy cloud products.

### PR DESCRIPTION
**Pull request by neallred**
#### Summary
When webapp asks, include request for legacy cloud products.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-38957

#### Related changes
[enterprise](https://github.com/mattermost/enterprise/pull/1075)
[webapp](https://github.com/mattermost/mattermost-webapp/pull/9032)
[customer-web-server](https://github.com/mattermost/customer-web-server/pull/577)

#### Release Note

```release-note
Add query param to get cloud products endpoint to allow webapp to request for legacy cloud products.
```

### CODE REVIEW

File: api4/cloud.go [hunk 1]
```diff
@@ -138,7 +138,11 @@ func getCloudProducts(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	products, err := c.App.Cloud().GetCloudProducts(c.AppContext.Session().UserId)
+	query := r.URL.Query()
+	_, includeLegacyProducts := query["include_legacy"]
```
Comment (marianunez) [inline api4/cloud.go:142]: Do we need to check `includeLegacyProducts` for empty or the actual `true` string?

`includeLegacyProduct := r.URL.Query().Get("include_legacy") == "true"`

**Review by marianunez — COMMENTED** [review]
PR title probably needs to be updated :)

File: api4/cloud.go (hunk 1 above)
Comment (neallred) [reply api4/cloud.go:142]: Yeah, we should.

**Review by marianunez — APPROVED** [review]

**Review by AGMETEOR — APPROVED** [review]
LGTM!

**Comment by AGMETEOR** [comment]
@neallred all code looks good. Just that release note looks a bit too abstract. Maybe we can document that the endpoint can now take a query param?

**Comment by pablovelezvidal** [comment]
> @neallred all code looks good. Just that release note looks a bit too abstract. Maybe we can document that the endpoint can now take a query param?

+1

**Review by nickmisasi — APPROVED** [review]

**Pull request merged**